iac-extensions/terraform/src/main/java/org/sonar/iac/terraform/checks/azure/AzurePublicNetworkAccessCheckPart.java

```diff
@@ -52,21 +49,44 @@ protected void registerResourceChecks() {
       "azurerm_postgresql_server",
       "azurerm_redis_cache",
       "azurerm_search_service",
-      "azurerm_synapse_workspace");
-    register(AzurePublicNetworkAccessCheckPart::checkPublicNetworkVariant, "azurerm_data_factory", "azurerm_purview_account");
+      "azurerm_synapse_workspace"), reportEnabledPublicIp("public_network_access_enabled"));
+    resourceConsumer(List.of("azurerm_data_factory", "azurerm_purview_account"), reportEnabledPublicIp("public_network_enabled"));
+
+    resourceConsumer("azurerm_application_gateway", reportPublicIpConfiguration("frontend_ip_configuration"));
+    resourceConsumer("azurerm_network_interface", reportPublicIpConfiguration("ip_configuration"));
+
+    resourceConsumer(List.of("azurerm_dev_test_linux_virtual_machine", "azurerm_dev_test_windows_virtual_machine"),
+      resource -> resource.attribute("disallow_public_ip_address")
+        .reportOnFalse(NETWORK_ACCESS_MESSAGE)
+        .reportAbsence(OMITTED_MESSAGE));
+
+    resourceConsumer("azurerm_dev_test_virtual_network",
+      resource -> resource.block("subnet").ifPresent(
+        subnet -> subnet.attribute("use_public_ip_address")
+          .reportUnexpectedValue("Deny", NETWORK_ACCESS_MESSAGE)
+          .reportAbsence(OMITTED_MESSAGE)));
```

Comment:
I think that's not what the ticket specifies but it's not very explicit:
According to the [reference](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/dev_test_virtual_network), `subnet` is optional but `use_public_ip_address` is not.
So I suppose the ticket's intention is to raise an issue when `subnet` is missing, not when `use_public_ip_address` is missing.

Reply:
I agree and check with AppSec